be/src/exec/base_scanner.cpp

```diff
@@ -332,7 +337,8 @@ Status BaseScanner::_materialize_dest_block(vectorized::Block* dest_block) {
             for (int i = 0; i < rows; ++i) {
                 if (filter_map[i] && nullable_column->is_null_at(i)) {
                     if (_strict_mode && (_src_slot_descs_order_by_dest[dest_index]) &&
-                        !_src_block.get_by_position(dest_index).column->is_null_at(i)) {
+                        !_src_block.get_by_position(_dest_slot_to_src_slot_index[dest_index])
```

Comment:
Better use `_dest_slot_to_src_slot_index.find` to instead `operator[]`, and add a check for not found situation

Reply:
If `_src_slot_descs_order_by_dest[dest_index]` is not nullptr, then the dest column has a src column, and `dest_index` exists in `_dest_slot_to_src_slot_index`.